tests/Unit/AutoRouteManagerTest.php

```diff
@@ -11,154 +11,360 @@
+                        'existsInDatabase' => false,
+                        'withSameContent' => false,
+                        'expectedUri' => '/foo/bar',
+                    ],
+                ],
+            ],
+            'a single localized route' => [
+                [
+                    [
+                        'generatedUri' => '/foo/bar',
+                        'locale' => 'fr',
+                        'existsInDatabase' => false,
+                        'withSameContent' => false,
+                        'expectedUri' => '/foo/bar',
+                    ],
+                ],
+            ],
+            'two routes' => [
+                [
+                    [
+                        'generatedUri' => '/foo/bar',
+                        'locale' => null,
+                        'existsInDatabase' => false,
+                        'withSameContent' => false,
+                        'expectedUri' => '/foo/bar',
+                    ],
+                    [
+                        'generatedUri' => '/bar/baz',
+                        'locale' => null,
+                        'existsInDatabase' => false,
+                        'withSameContent' => false,
+                        'expectedUri' => '/bar/baz',
+                    ],
+                ],
+            ],
+            'two localized routes' => [
+                [
+                    [
+                        'generatedUri' => '/foo/bar',
+                        'locale' => 'en',
+                        'existsInDatabase' => false,
+                        'withSameContent' => false,
+                        'expectedUri' => '/foo/bar',
+                    ],
+                    [
+                        'generatedUri' => '/bar/baz',
+                        'locale' => 'fr',
+                        'existsInDatabase' => false,
+                        'withSameContent' => false,
+                        'expectedUri' => '/bar/baz',
+                    ],
+                ],
+            ],
+            'a single route conflicting with a persisted one' => [
+                [
+                    [
+                        'generatedUri' => '/foo/bar',
+                        'locale' => null,
+                        'existsInDatabase' => true,
+                        'withSameContent' => false,
+                        'expectedUri' => '/foo/bar-resolved',
+                    ],
+                ],
+            ],
+            'a single localized route conflicting with a persisted one' => [
+                [
+                    [
+                        'generatedUri' => '/foo/bar',
+                        'locale' => 'fr',
+                        'existsInDatabase' => true,
+                        'withSameContent' => false,
+                        'expectedUri' => '/foo/bar-resolved',
+                    ],
+                ],
+            ],
+            'two conflicting routes' => [
```

Comment:
This description might be modified to be clearer. It's the use case of #88 : two routes in the same collection which are in conflict but not present in the database. It might be "two routes where the second one conflicts with the first one".